erts/emulator/test/jit_SUITE.erl

```diff
@@ -180,6 +180,48 @@ annotate(Config) ->
                     [Symbol, Anno])
     end.
 
+run_jmsingle_test(Param, ExpectSuccess, ErrorMsg) ->
+    Cmd = "erl +JMsingle " ++ Param ++ " -noshell " ++
+        "-eval 'io:format(\"All is well~n\"),erlang:halt(0).'",
+    Result = os:cmd(Cmd),
+    SuccessfulEmulatorStart =
+        case Result of
+            "All is well" ++ _ ->
+                true;
+            _ ->
+                Error = "Failed to allocate executable+writable memory",
+                case string:find(Result, Error) of
+                    nomatch -> false;
+                    _ -> internal_error
+                end
+        end,
+    case SuccessfulEmulatorStart of
+        ExpectSuccess ->
+            ok;
+        _ ->
+            ct:fail(ErrorMsg)
+    end.
+
+jmsingle(_Config) ->
+    %% Smoke test to check that the emulator starts with the +JMsingle
+    %% true/false option and fails with a non-boolean, that is, we
+    %% parse the command line correctly.
+    case erlang:system_info(system_architecture) of
+        "aarch64-apple" ++ _ ->
+            %% +JMsingle true does not work on macOS running
+            %% on Apple Silicon computers.
```

Comment:
`os:type() == {unix,netbsd}` has the same issue.

Reply:
Does the allocator error out at creation time or do we have to check for the os in `JitAllocator::create_allocator()` just like on the mac?

Reply:
It errors out at creation time. So the only thing we need to fix is the testcase.